src/OrbitBase/ThreadPool.cpp

```diff
@@ -197,4 +205,18 @@ std::shared_ptr<ThreadPool> ThreadPool::Create(
                                           std::move(run_action));
 }
 
+std::shared_ptr<ThreadPool> ThreadPool::GetGlobalWorkerThreadPool() {
+  static const unsigned number_of_logical_cores = std::thread::hardware_concurrency();
+  // std::thread::hardware_concurrency may return 0 on unsupported platforms but that shouldn't
+  // occur on standard Linux or Windows.
+  ORBIT_CHECK(number_of_logical_cores > 0);
+
+  static std::shared_ptr<ThreadPool> core_count_sized_thread_pool = orbit_base::ThreadPool::Create(
+      /*thread_pool_min_size=*/number_of_logical_cores,
+      /*thread_pool_max_size=*/number_of_logical_cores, /*thread_ttl=*/absl::Seconds(1),
+      /*run_action=*/[](const std::unique_ptr<Action>& action) { action->Execute(); });
+
+  return core_count_sized_thread_pool;
+}
```

Comment:
The thread pool should probably be created during process startup and not when `GetGlobalWorkerThreadPool` is called the first time. http://go/stadia-orbit-default-executor for how it's done internally.

Reply:
Added `void ThreadPool::InitializeDefaultThreadPool()` which is called on `OrbitApp` creation. Note that as is done internally, if we don't explicitly initialize or set the default thread pool, it will be created on the first call to `ThreadPool::GetDefaultThreadPool()`.